parser/src/token.rs

```diff
@@ -7,6 +7,7 @@ use self::Error::*;
 
 #[derive(Clone, PartialEq, Debug)]
 pub enum Token<'input> {
+    ShebangLine(String),
```

Comment:
You can use `&'input str` here for a (negligble) performance increase (more just to e consistent with the other variants).

Or thinking about it, since only the shell executing the file needs to know about it the variant could be empty instead. *Maybe* it could be useful in the future but do whatever seems simplest.